### PR TITLE
Add Apply and Update to Developer Guide

### LINKED ISSUE #112: Add Apply and Update feature to Developer Guide
# Add Apply and Update feature to Developer Guide
Building on top of #105 
due: 24 Oct 2024
## Changes
* Add Apply feature implementation to Developer Guide
* Add Update feature implementation to Developer Guide
* Add UML diagrams for Apply and Update feature

### PR DESCRIPTION
**Pull request by blanklogic**
Resolves #112

## ApplyCommandSequence.puml
ApplyCommandSequence

## UpdateCommandSequence.puml
UpdateCommandSequence

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (Shux347) [inline docs/DeveloperGuide.md:1]: The documentation you’ve added is very clear and concise. Great job!

**Review by Shux347 — APPROVED** [review]

File: docs/diagrams/ApplyCommandSequence.puml
Comment (uniqly) [inline docs/diagrams/ApplyCommandSequence.puml:1]: A preview on how this looks would be appreciated. 😄

File: docs/diagrams/UpdateCommandSequence.puml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+UpdateCommand -> Application: updateStatus(newStatus)
+Application --> UpdateCommand: updatedApplication
+
+UpdateCommand -> Company: updateApplication(updatedApplication)
+Company --> UpdateCommand: updatedCompany
+
+UpdateCommand -> Model: setCompany(targetCompany, updatedCompany)
+UpdateCommand -> Model: commitAddressBook()
+
+@enduml
```
Comment (uniqly) [inline docs/diagrams/UpdateCommandSequence.puml:19]: Same as above 🐱

**Review by uniqly — APPROVED** [review]
Great effort was put into the documentation. LGTM! 👍 

Would love to see a preview though (although not a strict requirement)

**Pull request merged**